main_old.py

```diff
@@ -0,0 +1,107 @@
+import json
+from pathlib import Path
+from dataclasses import dataclass, asdict, field
+from datetime import datetime
+import uuid
+import typer
+
+app = typer.Typer()
+
+DB_FILE = Path("journal.json")
+DB_FILE.touch(exist_ok=True)
+
+
+@dataclass(slots=True)
+class JournalEntry:
+
+    title: str
+    content: str
+    date: str
+    tags: list[str] = field(default_factory=list)
+    uuid: str = field(default_factory=lambda: uuid.uuid4().hex)
+
+
+def load_entries() -> list[JournalEntry]:
+    """Load journal entries from the JSON file."""
+
+    try:
+        with open(DB_FILE, "r") as f:
+            data = json.load(f)
+            return [
+                JournalEntry(
+                    title=entry.get("title", "Untitled"),
+                    content=entry.get("content", ""),
+                    date=entry.get("date", "Unknown date"),
+                    tags=entry.get("tags", [])
+                    uuid=entry.get("uuid", uuid.uuid4().hex),
```

Comment:
Seems you do this in default factory already, so might be able to leave off this default